core/src/test/java/org/elasticsearch/common/xcontent/builder/XContentBuilderTests.java

```diff
@@ -260,4 +263,36 @@ public void testCopyCurrentStructure() throws Exception {
 
         assertThat(i, equalTo(terms.size()));
     }
+
+    @Test
+    public void testHandlingOfPath() throws IOException {
+        Path path = PathUtils.get("path");
```

Comment:
I made it more explicit in the test that the value returned by toString() is used. Anything more would be testing the behavior of the toString() method of a particular Path implementation, which is not desirable I think. It could also make the test operating system specific.

I still included a relative path example, as it tests a Path with multiple components, which does highlight the need to escape the string value, at least when running the test on windows, where the separator is the backslash.